### ISSUE TITLE
Project missing unit tests and CI checks on PRs

### ISSUE THREAD

**Issue by DenysVuika** [kuhnroyal/mat-datetimepicker#94]
From what I see there are some problems with the quality control:
- the project is missing unit tests
- the default `yarn test` command fails, broken tests
- the Travis configuration is empty and has no tests

Expected behavior:
- there should be at least basic suite of tests running on each PR

**Comment by DenysVuika** [comment]
I can help with fixing this issue

**Comment by bbortt** [comment]
Feel free to add tests. What's not exactly true is that
> the Travis configuration is empty and has no tests

See the [.travis.yml in the master branch](https://github.com/kuhnroyal/mat-datetimepicker/blob/master/.travis.yml). However I'm not sure if it was propagated to the legacy branches. Maybe not :D

**Comment by bbortt** [comment]
I'm thinking about a fork in order to have full control over the repository.. Thoughts @kuhnroyal ?
We could update the `README.md` with the new location and archive this one. In the active fork we hat-tip this as the original (but no longer maintained) one (in addition to the fork-mark).
cc: @DenysVuika

**Comment by bbortt** [comment]
there is progress in terms of ci. thanks be the winter vacations..

I've migrated from Travis CI to GitHub actions. this includes automatic test and build for all branches. automatic version increase based on conventional commits. and automatic releases on top of that.
in addition, I've introduced automatic code formatting and basic linting.
those improvements together make the repository more attractive for contributors and ease the release lifecycle for maintainers.

the project still lacks tests, but forcing people to write them for existing code is not something we should go for, in my opinion. better to ask for tests in PRs when features or bugfixes are implemented. that way we can continuously increase coverage.
still, feel free to add tests @DenysVuika 😉 

closing this for now.
